pandas/tests/frame/test_dtypes.py

```diff
@@ -372,6 +372,76 @@ def test_astype_str(self):
             expected = DataFrame(['1.12345678901'])
             assert_frame_equal(result, expected)
 
+    def test_astype_dict(self):
```

Comment:
Is there a more elegant way to handle the `Series` case than just adding a new `if` statement to handle the `Series` explicitly? The above implementation doesn't work for `Series` because the `df[label]` accessor would try to access rows in the `Series`, and I think we want to map the keys in the dict to the `Series.name` property.